packages/EasyBankFiles/src/Parsers/DirectEntry/Parser.php

```diff
@@ -75,155 +87,240 @@ public function getErrors(): array
         $code = $line[0] ?? self::EMPTY_LINE_CODE;
-        $lineLength = \strlen($line);
-
-        if ($code === self::HEADER && $lineLength >= self::MIN_HEADER_LINE_LENGTH) {
-            $this->header = $this->processHeader($line);
 
-            return;
+        if ($this->checkMinLength($line, $code)) {
+            $lineProcessingResult = match ($code) {
+                self::RECORD_TYPE_HEADER => $this->processHeader($line),
+                self::RECORD_TYPE_TRAILER => $this->processTrailer($line),
+                self::RECORD_TYPE_TRANSACTION_PAYMENT => $this->processTransactionTypePayment($line),
+                self::RECORD_TYPE_TRANSACTION_RETURN => $this->processTransactionTypeReturn($line),
+                self::RECORD_TYPE_TRANSACTION_REFUSAL => $this->processTransactionTypeRefusal($line),
+                default => $this->isFinalLine($line)
+            };
+
+            if ($lineProcessingResult === true) {
+                return;
+            }
         }
 
-        if ($code === self::TRAILER && $lineLength >= self::MIN_TRAILER_LINE_LENGTH) {
-            $this->trailer = $this->processTrailer($line);
+        $this->errors[] = new Error(\compact('line', 'lineNumber'));
+    }
+
+    private function addTransactionToCurrentBatch(
+        TransactionTypePayment|TransactionTypeReturn|TransactionTypeRefusal $transaction
+    ): bool {
+        if (isset($this->currentBatch) === false || $this->currentBatch->hasTrailer()) {
+            $this->currentBatch = null;
 
-            return;
+            return false;
         }
+        $this->currentBatch->addTransaction($transaction);
 
-        if (
-            ($code === self::TRANSACTION_1 || $code === self::TRANSACTION_2) &&
-            $lineLength >= self::MIN_TRANSACTION_LINE_LENGTH
-        ) {
-            $this->transactions[] = $this->processTransaction($line);
+        return true;
+    }
 
-            return;
-        }
+    private function checkMinLength(string $line, string $code): bool
+    {
+        return isset(self::MIN_RECORD_LENGTH[$code]) === false || \strlen($line) >= self::MIN_RECORD_LENGTH[$code];
+    }
 
-        $this->errors[] = new Error(\compact('line', 'lineNumber'));
+    private function isFinalLine(string $line): bool
+    {
+        return (\trim($line) === self::FINAL_LINE || \trim($line, '9') === '') &&
+            ($this->currentBatch === null || $this->currentBatch->hasTrailer());
+    }
+
+    /**
+     * Parse transaction attributes are the same for each transaction type.
+     *
+     * @return array<mixed>
+     *
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     */
+    private function parseCommonTransactionAttributes(string $line): array
+    {
+        /** @var string|false $bsb */
+        $bsb = \substr($line, 1, 7);
+        /** @var string|false $accountNumber */
+        $accountNumber = \substr($line, 8, 9);
+        /** @var string|false $txnCode */
+        $txnCode = \substr($line, 18, 2);
```

Comment:
What about renaming to `transactionCode` to be consistent with docs and `\EonX\EasyBankFiles\Generators\Aba\Objects\Transaction`?

Reply:
I would like to add a point regarding this and other renaming suggestions that are motivated by docs consistency reasons. All they are really good suggestions. But.
First of all we have to keep the naming consistent in the context of this package. In other words if there is any naming conflict between existing classes after changing the naming we should satisfy the package consistency rather than the docs.
Second point - let's try to understand reasons for the current naming. For example, why there is `Header` if it's `DescriptiveRecord` in the docs.

After we have complete understanding for these points, we can make a decision about renaming.

Reply:
Based on my experience working with this package - when I have some questions, nobody doesn't know anything.

We need it one or two times in a year, and this becomes painful to investigate something.

So, I believe parses must be identical to the DirectEntry docs.

Now we create a brand new parser and may use any naming.

Later we will rewrite other parsers and generators to be consistent with DE docs and each other.

@akl-95, @alexandrtspl wdyt?

Reply:
If we plan to rewrite other parsers I agreed

Reply:
txnCode renamed